### ISSUE TITLE
BottomNavigationBar is not accessible

### ISSUE THREAD

**Issue by goderbauer** [flutter/flutter#14375]

**Comment by goderbauer** [comment]
@yjbanov This is a good example where our traversal order is way off on iOS:

```dart
import 'package:flutter/material.dart';

void main() => runApp(new MyApp());

class MyApp extends StatelessWidget {
  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return new MaterialApp(
      title: 'Flutter Demo',
      theme: new ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: new BottomNavDemo(),
    );
  }
}

class BottomNavDemo extends StatefulWidget {
  @override
  State<BottomNavDemo> createState() {
    return new _BottomNavDemoState();
  }
}

class _BottomNavDemoState extends State<BottomNavDemo> {
  // The current page we're looking at.
  int currentPageIndex = 0;

  // How many pages from `pageTitles` we show.
  int shownPages = 5;

  List<String> pageTitles = <String>['Foo', 'Bar', 'Baz', 'Doge', 'Pupper'];

  // The minimum number of pages we'll show.
  static const minPages = 3;
  static const padding = const Padding(
    padding: const EdgeInsets.symmetric(vertical: 8.0),
  );

  @override
  Widget build(BuildContext context) {
    return new Scaffold(
      appBar: new AppBar(
        title: new Text(
          'Bottom Navigation',
          style: Theme.of(context).textTheme.title,
        ),
        elevation: 0.0,
        backgroundColor: Colors.white,
      ),
      bottomNavigationBar: new BottomNavigationBar(
        items: _buildItems(context),
        type: BottomNavigationBarType.fixed,
        currentIndex: currentPageIndex,
        onTap: _selectPage,
      ),
      body: new Center(
        child: new Text('Page ${pageTitles[currentPageIndex]}'),
      ),
    );
  }

  List<BottomNavigationBarItem> _buildItems(BuildContext context) {
    return pageTitles.take(shownPages).map((pageTitle) {
      return new BottomNavigationBarItem(
        icon: new Icon(Icons.star),
        title: new Text(pageTitle),
        backgroundColor: Colors.white,
      );
    }).toList();
  }

  void _selectPage(int value) {
    setState(() {
      currentPageIndex = value;
    });
  }
}
```

**Comment by goderbauer** [comment]
@gspencergoog The out-of-the-box computed traversal order for Android also fails on this one: If the middle tab of 5 (number 3) is selected, the computed order is Tab 1, 2, 4, 5, 3. This looks like something that should be out-of-the-box correct without requiring fine tuning of traversal order.
